src/components/PopupWithForm/PopupWithForm.js

```diff
@@ -1,18 +1,32 @@
 import React from "react";
+import "../PopupWithForm/PopupWithForm.css";
+import { useFormWithValidation } from '../../Hooks/useFormWithValidation';
+import { Link } from 'react-router-dom';
+
+function PopupWithForm({ isOpen, children, onClose, onSubmit, formTitle, buttonText, linkClickHandler, linkText }) {
+    const { isValid } = useFormWithValidation();
 
-function PopupWithForm({
-    isOpen,
-    children,
-}) {
     return (
```

Comment:
The close icon should be also inside `PopupWithForm` not to duplicate it for each modal with a form

And you need to attach it with `absolute` to the `popup__container` or `popup__content-container`

Reply:
Done